### PR TITLE
actions: periodically generate site data

### PR DESCRIPTION
**Pull request by EricFromCanada**
[As suggested](https://github.com/Homebrew/homebrew-core/pull/53939#issuecomment-620959807), assume the task of periodically regenerating site data from changes in homebrew-core, homebrew-cask, and linuxbrew-core. Will need `secrets.ANALYTICS_JSON_KEY` and `secrets.ANALYTICS_JSON_KEY_B64` added to this repo for analytics processing to work.

### CODE REVIEW

**Comment by issyl0** [comment]
All the memories are coming back now. 😄

File: .github/workflows/scheduled.yml [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+          git add _data/cask api/cask cask
+
+          if ! git diff --no-patch --exit-code HEAD -- _data/cask api/cask cask; then
+            git commit -m "cask: updates from homebrew/cask" _data/cask api/cask cask
+            git fetch
+            git rebase origin/master
+            git push
+          fi
+
+  generate-linux:
```
Comment (MikeMcQuaid) [inline .github/workflows/scheduled.yml:66]: Thoughts on making this a matrix job so it can share more code?

Comment (MikeMcQuaid) [reply .github/workflows/scheduled.yml:66]: (check the homebrew-core or homebrew-test-bot repos for prior art for e.g. the setup)

Comment (EricFromCanada) [reply .github/workflows/scheduled.yml:66]: Any ideas for a container to apply to only one OS in a matrix? [No luck](https://github.com/EricFromCanada/formulae.brew.sh/actions/runs/94490949) with that so far…

Comment (MikeMcQuaid) [reply .github/workflows/scheduled.yml:66]: No ideas sorry 😭

**Comment by MikeMcQuaid** [comment]
Looks great, merge when ready @EricFromCanada. I think it might be nice to add a "generated at" timestamp or something to the page footer?

**Comment by EricFromCanada** [comment]
One odd behaviour I'm seeing is updates for Homebrew/linuxbrew-core data [translating requirements into Casks](https://github.com/EricFromCanada/formulae.brew.sh/commit/3e5552d67cb0236baaa0cd29908b01de9d7676e5), which should be a Mac-only behaviour.

**Comment by EricFromCanada** [comment]
> My guess: openjdk is installed on the Actions node through the system package manager, and abcl uses depends_on :java , which picks up the Requirement from the system instead of fulfilling it via a Homebrew formula. (@jonchang)

Will leave the question of whether a Linux formula's JSON output should be referencing casks for another time.

**Comment by EricFromCanada** [comment]
> add a "generated at" timestamp or something to the page footer?

@MikeMcQuaid By this do you mean the footer of each individual formula's web page, or just other select pages?

**Comment by MikeMcQuaid** [comment]
> @MikeMcQuaid By this do you mean the footer of each individual formula's web page, or just other select pages?

I think even just a site-wide footer on the homepage (and in JSON files) would be useful.

**Comment by jonchang** [comment]
Nice work on this!

**Comment by MikeMcQuaid** [comment]
Thanks @EricFromCanada and for cleaning up all the existing jobs too 😻

**Pull request merged**